src/extension/panel.ts

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { autorun, IArraySplice, observable, observe } from 'mobx';
+import { Log, Result } from 'sarif';
+import { commands, ExtensionContext, TextEditorRevealType, Uri, ViewColumn, WebviewPanel, window, workspace } from 'vscode';
+import { filtersColumn, filtersRow, ResultId, _Region } from '../shared';
+import { Baser } from './baser';
+import { loadLogs } from './loadLogs';
+import { regionToSelection } from './regionToSelection';
+import { Store } from './store';
+
+export class Panel {
+    private title = 'SARIF Result'
+    @observable private panel = null as WebviewPanel | null
```

Comment:
>private panel = null as WebviewPanel | null [](start = 16, length = 43)

This would be far easier to read as
private panel: WebviewPanel | null = null;

Also, typically you should only use undefined in typescript.

Reply:
When you say " typically you should only use undefined in typescript" is that your opinion are you referencing something?

I struggle with this. Understand the subtle semantics between the two, but I don't think I've ever seen a case where we couldn't always use just undefined. However, all the native API's (and so many external libraries) use `null` to denote absence when the pointer is a reference type. I feel the ideal is lost cause, so I aim for the next best thing: consistency with the predominant pattern.

---
In reply to: [441649046](https://github.com/microsoft/sarif-vscode-extension/pull/282#discussion_r441649046) [](ancestors = 441649046)

Reply:
Changed this to colon notation.

---
In reply to: [442430959](https://github.com/microsoft/sarif-vscode-extension/pull/282#discussion_r442430959) [](ancestors = 442430959,441649046)